src/Simplex/Messaging/Agent.hs

```diff
@@ -2133,7 +2133,7 @@ cleanupManager c@AgentClient {subQ} = do
       step <- asks $ cleanupStepInterval . config
       liftIO $ threadDelay step
     -- we are catching it to avoid CRITICAL errors in tests when this is the only remaining handle to active
-    waitActive a = liftIO (E.tryAny . atomically $ waitUntilActive c) >>= either (\_ -> pure ()) (\_ -> void a)
+    waitActive a = liftIO (E.tryAny $ waitUntilActive c) >>= either (\_ -> pure ()) (\_ -> void a)
```

Comment:
this is suspicious - why does it need "try"?

Reply:
maybe some older wrong refactoring? not sure. we'd only want to try separate steps and proceed onto next

Reply:
no, waitUntilActive can't throw

Reply:
ah - there is some comment...